app/models/topic_user.rb

```diff
@@ -381,6 +377,24 @@ def update_last_read(user, topic_id, post_number, new_posts_read, msecs, opts =
       end
     end
 
+    private
+
+    def publish_read(topic_id:, post_number:, user:, notification_level: nil, private_message:)
```

Comment:
I don't get why this is necessary when both classes have the same method and parameters?

Reply:
This method here is to decide which class of topic tracking state it should use. I didn't want to duplicate this logic in multiple places in the class.

Reply:
Oh right of course not sure what I was thinking with this comment...carry on!